code/controllers/subsystems/processing/odyssey.dm

```diff
@@ -0,0 +1,225 @@
+		log_and_message_admins(SPAN_DANGER(FONT_HUGE("CRITICAL ERROR: NO SITUATIONS ARE AVAILABLE FOR THIS SECTOR! CHANGE THE GAMEMODE MANUALLY!")))
+		return FALSE
+
+	scenario = pickweight(possible_scenarios)
+	gamemode_setup()
+	/// Now that we actually have an odyssey, the subsystem can fire!
+	can_fire = TRUE
+	horizon = SSshuttle.ship_by_type(/obj/effect/overmap/visitable/ship/sccv_horizon)
+	return TRUE
+
+/**
+ * This proc overrides the gamemode variables for the amount of actors.
+ * Note that Storytellers spawn through a ghost role.
+ */
+/datum/controller/subsystem/odyssey/proc/gamemode_setup()
+	SHOULD_CALL_PARENT(TRUE)
+	var/datum/game_mode/odyssey/ody_gamemode = GLOB.gamemode_cache["odyssey"]
+	if(scenario)
+		ody_gamemode.required_players = scenario.min_player_amount
+		ody_gamemode.required_enemies = scenario.min_actor_amount
+
+/**
+ * This is the proc that should be used to set the odyssey away site.
+ */
+/datum/controller/subsystem/odyssey/proc/set_scenario_site(datum/map_template/ruin/away_site/site)
+	if(!istype(site))
+		return
+
+	scenario_site = site
+
+/**
+ * Adds an actor to the subsystem actor list.
+ */
+/datum/controller/subsystem/odyssey/proc/add_actor(mob/living/carbon/human/H)
+	LAZYDISTINCTADD(actors, H)
+
+/**
+ * Adds a storyteller to the subsystem storyteller list.
+ */
+/datum/controller/subsystem/odyssey/proc/add_storyteller(mob/abstract/ghost/storyteller/S)
+	LAZYDISTINCTADD(storytellers, S)
+
+/**
+ * Removes an actor from the subsystem actor list.
+ */
+/datum/controller/subsystem/odyssey/proc/remove_actor(mob/living/carbon/human/H)
+	LAZYREMOVE(actors, H)
+
+/**
+ * Removes a storyteller from the subsystem storyteller list.
+ */
+/datum/controller/subsystem/odyssey/proc/remove_storyteller(mob/abstract/ghost/storyteller/S)
+	LAZYREMOVE(storytellers, S)
+
+/datum/controller/subsystem/odyssey/ui_state()
+	return GLOB.always_state
+
+/datum/controller/subsystem/odyssey/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "OdysseyPanel", "Odyssey Panel")
+		ui.open()
+
+/datum/controller/subsystem/odyssey/ui_data(mob/user)
+	var/list/data = list()
+	if(scenario)
+		data["scenario_name"] = SSodyssey.scenario.name
+		data["scenario_desc"] = SSodyssey.scenario.desc
+		data["scenario_canonicity"] = SSodyssey.scenario.scenario_type == SCENARIO_TYPE_CANON ? "Canon" : "Non-Canon"
+		data["is_storyteller"] = isstoryteller(user)
+
+		if(length(scenario.roles))
+			data["scenario_roles"] = list()
+			for(var/role_singleton in scenario.roles)
+				var/singleton/role/scenario_role = GET_SINGLETON(role_singleton)
+				data["scenario_roles"] += list(
+					list(
+						"name" = scenario_role.name,
+						"desc" = scenario_role.desc,
+						"outfit" = "[scenario_role.outfit]",
+						"type" = scenario_role.type
+					)
+				)
+	return data
+
+/datum/controller/subsystem/odyssey/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
```

Comment:
The return value of parent is significant and should be checked

Reply:
there isn't a parent proc

Reply:
If there wasn't a parent proc, the linter would not let you call parent, it would error out, and it would have the name 

The parent proc does indeed exist, that's under /datum, and returns TRUE if the UI doesn't exist or isn't interactive, which is why the return value of parent is significant and should be checked